components/signup/Signup.tsx

```diff
@@ -0,0 +1,157 @@
+import Input from "../form/Input";
+import axios from 'axios'
+import Link from "next/link";
+import { FcGoogle } from 'react-icons/fc'
+import { FaFacebookSquare } from 'react-icons/fa'
+
+const Signup: React.FC = () => {
+
+  // States
+  const [username, setUsername] = useState<string>('')
+  const [orgName, setOrgName] = useState<string>('')
+  const [email, setEmail] = useState<string>('')
+  const [password, setPassword] = useState<string>('')
+  
+  const [isUsernameValid, setIsUsernameValid] = useState<boolean>(false)
+  const [isOrgNameValid, setIsOrgNameValid] = useState<boolean>(false)
+  const [isEmailValid, setIsEmailValid] = useState<boolean>(false)
+  const [isPasswordValid, setIsPasswordValid] = useState<boolean>(false)
+  
+  const [usernameError, setUsernameError] = useState<string>('')
+  const [orgNameError, setOrgNameError] = useState<string>('')
+  const [emailError, setEmailError] = useState<string>('')
+  const [passwordError, setPasswordError] = useState<string>('')
+
+  // RegEx
+  const usernameRegex = /^[a-zA-Z]{3,12}$/;
+  const orgNameRegex = /^[a-zA-Z0-9\s]+$/;
+  const emailRegex = /^([^\s@]+@[^\s@]+\.[^\s@]+)$/;
+  const passwordRegex = /^(?=.*[A-Z])(?=.*[0-9])[A-Za-z0-9]{8,}$/;
+
+  // Functions
+  const submitForm = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault()
+
+    // Username Validation
+    if(!usernameRegex.test(username)){
+        setUsernameError("Username can only contain letters and must be between 3 and 12 characters long.")
+    }else{
+        setIsUsernameValid(true)
+        setUsernameError('')
+    }
+
+    // Organization Name Validation
+    if(!orgNameRegex.test(orgName)){
+      setOrgNameError("Organization Name should only contains letters and numbers")
+    }else{
+      setIsOrgNameValid(true)
+      setOrgNameError('')
+    }
+
+    // Email validation
+    if(!emailRegex.test(email)){
+        setEmailError("Please enter a valid email address.")
+    }else{
+        setIsEmailValid(true)
+        setEmailError('')
+    }
+
+    // Password validation
+    if(!passwordRegex.test(password)){
+        setPasswordError("Please include at least one uppercase letter and a number. It must be at least 8 characters long.")
+    }else{
+        setIsPasswordValid(true)
+        setPasswordError('')
+    }
+
+
+    if(isUsernameValid && isOrgNameValid && isEmailValid && isPasswordValid){
+        try {
+            const response = await axios.post("", 
+                {username, orgName, email, password},
+                {
+                    headers: {
+                        "Content-Type" : "application/json"
+                    }, 
+                    withCredentials: true
+                }
+                )
+            const data = await response.data
+            console.log(data)
+        } catch (error) {
+            console.log(error)
+        }
+    }
+  }
+
+const handleUsername = (e: React.ChangeEvent<HTMLInputElement>): void => {
```

Comment:
Remove 1 liner functions, just add the line directly where needed.

Reply:
completed